lib/semian/adapter.rb

```diff
@@ -35,7 +35,8 @@ def acquire_semian_resource(scope:, adapter:, &block)
         mark_resource_as_acquired(&block)
       end
     rescue ::Semian::OpenCircuitError => error
-      raise self.class::CircuitOpenError.new(semian_identifier, error.message)
+      last_error_message = semian_resource.circuit_breaker.last_error_message
+      raise self.class::CircuitOpenError.new(semian_identifier, "#{error.message} caused by #{last_error_message}")
```

Comment:
Could we perhaps hold onto the original exception object, and set it as the proper error cause? Then the full stack will show up in error reporting.

Reply:
Just to make sure I understand you well, you mean instead of raising `CircuitOpenError` we would raise for example `Net::Timeout` ?

Reply:
First time working with the `cause` method, so I might be missing something. It seems like I can't set the `cause` value easily. It seems like I need to raise the last error then catch it so that the `cause` value get sets to the `last_error`. I could, however, patch the `SemianError` class to add a `cause` method but I'm not sure if that's the right thing to do.

I've looked into [this](https://ruby-doc.org/core-2.5.0/Exception.html#method-i-cause) and [this](http://www.virtuouscode.com/2013/12/25/exception-causes-in-ruby-2-1/) to get a better idea of how the `cause` method works.

Reply:
We should find out how bugsang displays this before choosing to go with this method.

Reply:
I tested it on a small rails app and here's how it looks like:
![image](https://user-images.githubusercontent.com/5659685/46840529-63dd0980-cd78-11e8-95cd-af5f66dce7db.png)

I think it make sense and it's very clear. I'll go with this 👍

Reply:
@pushrax Let me know if the change makes sense!